core/variant.cpp

```diff
@@ -2849,7 +2849,7 @@ uint32_t Variant::hash() const {
 			return _data._bool ? 1 : 0;
 		} break;
 		case INT: {
-			return _data._int;
+			return uint32_t(_data._int);
```

Comment:
Doesn't this mean that several integers will have the same hash? (I remember commenting the same on an earlier PR about those... :)).

Reply:
Results are same both functions, but is only added explicit casting which suppress warnings
```
uint32_t pp(){
    return uint32_t(-1);
}
uint32_t rr(){
    return -1;
}
```

`_data._int` have type `int64_t` and returned is value `uint32_t` so yes a lot of ints should have same hash, but I think it's the result of a well thought-out decision and not an error (maybe if Godot 4.0 focuses more on 64-bit machines, returned value can be changed to uint64_t)


In previous PR I wanted to get properly value not by explicit casting but subtraction, which doesn't always work(it was broken in simpler words)

Reply:
Why do all of the hashing functions use `uint32_t`? Perhaps they should be changed to return `uint64_t` instead so that hash conflicts are less likely. Godot will target 64-bit systems as a priority, so it may be that some previous downsides no longer exist (still, 2x memory, and the math probably needs to be longer for a 64-bit hash), though I have no idea how frequently this code is used or how performance critical it is.

But yeah, qarmin is right, it still truncates to `uint32_t` anyway because that's what the method returns.